src/layer/place.js

```diff
@@ -309,9 +325,14 @@ export const country2 = {
   type: "symbol",
   paint: {
     "text-color": "#334",
-    "text-halo-blur": 1,
-    "text-halo-color": "rgba(255,255,255,1)",
-    "text-halo-width": 3.0,
+    "text-halo-blur": 0,
+    "text-halo-color": Color.backgroundFill,
+    "text-halo-width": [
+      'interpolate', ['linear'],
+      ['zoom'],
+      2, 2,
+      5, 3,
+    ],
```

Comment:
I noticed that the blocky artifacts in label halos were showing up more at lower zooms than higher ones.  This makes sense because the halo width was set to a fixed pixel value while the text size increased with the zoom level.  So 3 pixlels could be a perfect halo width at zoom 5 and higher where the font size is 17, but as you zoom out an the font size drops to 11, available space in the letter box also shinks and a 3 pixel halo starts to hit the edge and cause artifacts.  Replacing the fixed value with an interpolate expression seems to (mostly?) resolve the artifact issue.

I've also set the halo blur to zero as @1ec5 suggested so we can have a look at that as a possibility.  I think it may work nicely.  I'm sure bringing back a small blur would be viable if deemed necessary. It would probably need to be interpolated as well and the halo width would need to shrink a bit to avoid hitting the letter box edges and causing artifacts.

  Using an interpolate expression here instead of a fixed